source/particle/property/interface.cc

```diff
@@ -0,0 +1,333 @@
+        particle.set_n_property_components(n_property_components);
+        for (typename std::list<std_cxx11::shared_ptr<Interface<dim> > >::const_iterator
+             p = property_list.begin(); p!=property_list.end(); ++p)
+          {
+            (*p)->initialize_one_particle_property(particle.get_location(),
+                                                   solution,
+                                                   gradients,
+                                                   particle_properties);
+          }
+        particle.set_properties(particle_properties);
+      }
+
+      template <int dim>
+      void
+      Manager<dim>::update_one_particle (Particle<dim> &particle,
+                                         const Vector<double> &solution,
+                                         const std::vector<Tensor<1,dim> > &gradients) const
+      {
+        unsigned int property = 0;
+        for (typename std::list<std_cxx11::shared_ptr<Interface<dim> > >::const_iterator
+             p = property_list.begin(); p!=property_list.end(); ++p,++property)
+          {
+            (*p)->update_one_particle_property(positions[property],
+                                               particle.get_location(),
+                                               solution,
+                                               gradients,
+                                               particle.get_properties());
+          }
+      }
+
+      template <int dim>
+      UpdateTimeFlags
+      Manager<dim>::need_update () const
+      {
+        UpdateTimeFlags update = update_never;
+        for (typename std::list<std_cxx11::shared_ptr<Interface<dim> > >::const_iterator
+             p = property_list.begin(); p!=property_list.end(); ++p)
+          {
+            update = std::max(update,(*p)->need_update());
+          }
+        return update;
+      }
+
+      template <int dim>
+      unsigned int
+      Manager<dim>::get_n_property_components () const
+      {
+        return n_property_components;
+      }
+
+      template <int dim>
+      std::size_t
+      Manager<dim>::get_particle_size () const
+      {
+        return (n_property_components+dim) * sizeof(double) + sizeof(types::particle_index);
+      }
+
+      template <int dim>
+      const std::vector<std::pair<std::string, unsigned int> > &
+      Manager<dim>::get_data_info () const
+      {
+        return property_component_list;
+      }
+
+      template <int dim>
+      unsigned int
+      Manager<dim>::get_property_component_by_name(const std::string &name) const
+      {
+        // see if the given name is defined
+        if (property_position_map.find (name) != property_position_map.end())
+          return property_position_map.find(name)->second;
+        else
+          {
+            Assert(false,ExcMessage("The particle property manager was asked for "
+                                    "the property called: " + name +". This property does not exist "
```

Comment:
we typically write this as
```
  "...the property called <" + name + ">. ..."
```
This format has turned out to be useful because it clearly separates sentence and the string you quote. It also makes it clear if `name` for whatever reasons ends up having spaces at the front or back.